crate/server/src/core/cover_crypt/update_policy.rs

```diff
@@ -138,6 +170,123 @@ fn rotate_policy(
     Ok(policy)
 }
 
+/// Removes old rotations from the policy of the given Master Private Key
+/// and return it
+fn clear_old_rotations_policy(
+    private_key: &Object,
+    cover_crypt_policy_attributes_to_update: &[Attribute],
+) -> KResult<Policy> {
```

Comment:
After reading five times almost the same function, I wonder if it would not be better to take advantage of Rust's higher order functions:
```rust
fn mutate_policy_attributes(
    private_key: &Object,
    mutator: impl FnOnce(&mut Policy) -> KResult<()>,
) -> KResult<Policy> {
    let private_key_attributes = private_key.attributes()?;
    let mut policy = policy_from_attributes(private_key_attributes)?;
    mutator(&mut policy)?;
    Ok(policy)
}

fn add_policy_attribute(
    private_key: &Object,
    new_attributes: Vec<(Attribute, EncryptionHint)>,
) -> KResult<Policy> {
    mutate_policy_attributes(private_key, move |policy: &mut Policy| {
        for (attribute, encryption_hint) in new_attributes.into_iter() {
            policy.add_attribute(attribute, encryption_hint)?
        }
        Ok(())
    })
}

fn disable_policy_attributes(private_key: &Object, targets: &[Attribute]) -> KResult<Policy> {
    mutate_policy_attributes(private_key, move |policy: &mut Policy| {
        for attribute in targets {
            policy.disable_attribute(attribute.clone())?
        }
        Ok(())
    })
}
```